src/braket/devices/device.py

```diff
@@ -104,3 +110,36 @@ def status(self) -> str:
             str: The status of this Device
         """
         return self._status
+
+    def _validate_device_noise_model_support(self, noise_model: NoiseModel) -> None:
+        supported_noises = set(
+            supported_noise_pragma_to_noise[pragma].__name__
+            for pragma in self.properties.action[DeviceActionType.OPENQASM].supportedPragmas
+            if pragma in supported_noise_pragma_to_noise
+        )
+        noise_operators = set(noise_instr.noise.name for noise_instr in noise_model._instructions)
+        if noise_operators > supported_noises:
```

Comment:
Rather than doing a `>` comparison, you should explicitly check whether there is anything in `noise_operators` that isn't in `supported_noises`; it's conceivable that `noise_operators` contains something not in `supported_noises` while not being a strict superset:
```python
# This would pass, when it should actually faily
noise_operators = {1, 2, 3}
supported_noises = {2, 3, 4, 5}
```